fbpic/boundaries/boundary_communicator.py

```diff
@@ -532,7 +680,7 @@ def gather_grid( self, grid, root = 0):
         if self.rank == root:
             # Calculate global edges of the simulation box on root process
             zmin_global = grid.zmin + self.dz * \
-                            (self.n_guard - self.rank*self.Nz_domain)
+                (self.n_guard + self.n_damp - self.rank*self.Nz_domain)
```

Comment:
I would still calculate more explicitly: 
```python
nz_invalid = self.n_guard
if self.left_proc is None:
    nz_invalid += self.n_damp
zmin_global = grid.zmin + self.dz * (nz_invalid - self.rank*self.Nz_domain)
```
Mostly because root could be different than 0.